packages/frontend/src/pages/Admin.tsx

```diff
@@ -154,6 +281,118 @@ function ReportedRatings() {
+                        >
+                            Resume Audit
+                        </Button>
+                    )}
+                </div>
+
+                {/* Manual Resume Section */}
+                {(auditProgress.isPaused || auditProgress.nextCursor) && (
+                    <div className="mb-4 p-3 bg-yellow-50 border border-yellow-200 rounded">
+                        <h4 className="font-medium mb-2">Resume from Cursor:</h4>
+                        <div className="flex items-center gap-2 mb-2">
+                            <input
+                                type="text"
+                                value={resumeCursor}
+                                onChange={(e) => setResumeCursor(e.target.value)}
+                                placeholder="Enter cursor to resume from..."
+                                className="flex-1 px-3 py-2 border border-gray-300 rounded text-sm"
+                            />
+                            <Button
+                                type="button"
+                                onClick={resumeFromCursor}
+                                disabled={!resumeCursor.trim()}
+                                className="bg-blue-500 hover:bg-blue-600"
+                            >
+                                Resume
+                            </Button>
+                        </div>
+                        {auditProgress.nextCursor && (
+                            <div className="text-sm">
+                                <p className="text-gray-600 mb-1">Current cursor:</p>
+                                <code className="bg-gray-100 px-2 py-1 rounded text-xs break-all">
+                                    {auditProgress.nextCursor}
+                                </code>
+                                <button
+                                    type="button"
+                                    onClick={() => {
+                                        navigator.clipboard.writeText(
+                                            auditProgress.nextCursor || "",
+                                        );
+                                    }}
+                                    className="ml-2 text-blue-500 hover:text-blue-700 text-xs"
+                                >
+                                    Copy
+                                </button>
+                            </div>
+                        )}
+                    </div>
+                )}
+
+                {/* Progress Display */}
+                {(auditProgress.processedCount > 0 || auditProgress.isRunning) &&
+                    (() => {
+                        const progressPercent =
+                            auditProgress.totalProfessors > 0
+                                ? (auditProgress.processedCount / auditProgress.totalProfessors) *
+                                  100
+                                : 0;
```

Comment:
This could be simplified if the default value for `auditProgress.processedCount` was 0